qiskit_ibm_provider/jupyter/live_data_widget.py

```diff
@@ -0,0 +1,1385 @@
+        self._progress_bar.min = 0
+        self._progress_bar.max = 10
+        self._progress_bar.value = 10
+        self._percentage.value = "100%"
+        self._progress_bar.style = {"bar_color": CarbonColors.GREEN50.value}
+
+
+class JobInformationView:
+    """Job Information component"""
+
+    def __init__(self, job, backend) -> None:
+        self._backend = backend
+        self._progress_bar = ProgressBar()
+        self._progress_bar_widget = self._progress_bar.create_progress_bar()
+        self._information_view = self.create_job_information_view(job)
+
+    def widget(self) -> widgets.VBox:
+        """Create the Job Information widgets"""
+        box_layout = widgets.Layout(display="flex-inline", margin="0px")
+
+        return widgets.VBox(
+            children=(self._progress_bar_widget, self._information_view),
+            layout=box_layout,
+        )
+
+    def set_job(self, job) -> None:
+        """Fill the job information view
+
+        Args:
+
+            job (Dict with the fields 'id', 'liveDataEnabled' and 'creationDate'): the job to select
+
+        """
+        if not job:
+            return
+
+        # To get all the information needed, we request the job details to the API
+        # The information returned is the type of QiskitJob
+        qiskit_job = self._backend.provider().backend.retrieve_job(job_id=job["id"])
+        status = qiskit_job.status()
+        if status in [JobStatus.RUNNING, JobStatus.DONE]:
+            self.show_progress_bar()
+        elif status in [JobStatus.VALIDATING, JobStatus.INITIALIZING]:
+            self._progress_bar.reset_progress_bar()
+            self.show_progress_bar()
+        elif status in [JobStatus.ERROR]:
+            self._progress_bar.error_progress_bar()
+            self.show_progress_bar()
+        else:
+            self._progress_bar.reset_progress_bar()
+
+        self.information_view.value = self.job_information_content(qiskit_job)
+
+    # Views
+    def create_job_information_view(self, job) -> widgets.HTML:
+        """Create the job information view"""
+        content = self.job_information_content(job)
+        layout = widgets.Layout(display="flex", justify_content="flex-start")
+        self.information_view = widgets.HTML(value=content, layout=layout)
+        return self.information_view
+
+    def job_information_content(self, job) -> str:
+        """Create the job information content"""
+        content = "<table class='livedata-table'>"
+        content += """
+            <style>
+            .livedata-table {
+                border-collapse: collapse;
+                width: auto;
+                margin-left: 0px;
+            }
+            .livedata-table td {
+                text-align: left;
+                height: 6px;
+                min-width: 70px;
+                padding-left: 0px;
+                padding-right: 40px;
+                padding-top: 0px;
+                padding-bottom: 0px;
+            }
```

Comment:
Moving forward I would personally try to avoid that UI code makes its way into this Python client-library. Hard to keep up-to-date and maintain with production-grade quality .

Reply:
Yes we can create a separate package later on for UI related stuff. Just to clarify, we should still merge this PR when ready.